icelake/src/io/mod.rs

```diff
@@ -14,3 +13,26 @@ pub mod parquet;
 mod scan;
 pub mod writer_builder;
 pub use scan::*;
+
+use arrow_array::RecordBatch;
+use arrow_schema::SchemaRef;
+
+use crate::types::DataFileBuilder;
+use crate::Result;
+
+#[async_trait::async_trait]
+pub trait RecordBatchWriter: Send + 'static {
+    async fn write(&mut self, batch: RecordBatch) -> Result<()>;
+    async fn close(&mut self) -> Result<Vec<DataFileBuilder>>;
+}
+
+#[async_trait::async_trait]
+pub trait RecordBatchWriterBuilder: Send + Sync + Clone + 'static {
+    type R: RecordBatchWriter;
+    async fn build(self, schema: &SchemaRef) -> Result<Self::R>;
+}
+
+pub trait SingletonWriterStatus {
```

Comment:
This usage is a little weird to me, how about this:
```rust
pub trait SingleFileWriter: RecordBatchWriter {
   fn current_file(&self) -> String;
  fn current_row_num(&self) -> usize;
}
```

Reply:
nice!